src/BloomExe/web/ServerBase.cs

```diff
@@ -116,117 +74,122 @@ private bool StartWithExceptionHandling(out Exception error)
 			catch (Exception e)
 			{
 				error = e;
-				_listener = null;
 				return false;
 			}
 		}
 
 		private bool TryStart()
 		{
-			_listener = new HttpListener();
 			_listener.AuthenticationSchemes = AuthenticationSchemes.Anonymous;
 			_listener.Prefixes.Add(PathEndingInSlash);
 			_listener.Start();
-			//			_listener.BeginGetContext(new AsyncCallback(ListenerCallback), _listener);
 
-			_listenerThread = new Thread(HandleRequests);
 			_listenerThread.Start();
 
+			for (var i = 0; i < _workers.Length; i++)
+			{
+				_workers[i] = new Thread(WaitForRequests);
+				_workers[i].Start();
+			}
+
 			return GetIsAbleToUsePort();
 		}
 
-		private bool GetIsAbleToUsePort()
+		private void HandleRequests()
 		{
-			IsAbleToUsePort = false;
-			try
+			while (_listener.IsListening)
 			{
-				var x = new WebClientWithTimeout { Timeout = 3000 };
+				var context = _listener.BeginGetContext(ContextReady, null);
```

Comment:
Since we have a dedicated thread with nothing to do but wait for requests from the _listener, why do we need to use an async call to get them? Just so we can also terminate the loop on _stop?